controllers/certificate/certificate.go

```diff
@@ -0,0 +1,185 @@
+package certificate
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math"
+	"math/big"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	netutils "k8s.io/utils/net"
+)
+
+func generateSelfSignedCertificate(host string, alternateIPs []net.IP, alternateDNS []string, maxAge time.Duration) ([]byte, []byte, error) {
```

Comment:
Maybe it makes sense to split up the body of this function in two functions, e.g. createCaCert and createServerCert, to make it easier to follow its logic.

Reply:
I would say it make sense and I tried to split it, but there are many common variables used for all certificates, CA, private key and public certificate so I would keep it together like in: https://github.com/kubernetes/client-go/blob/3aa45779f2e5592d52edf68da66abfbd0805e413/util/cert/cert.go#L108